import.py

```diff
@@ -649,7 +652,63 @@ def write_to_file(cont: str, filename: str) -> None:
         f.write(cont)
 
 
+def prepend_file(filename: str, new_str: str) -> None:
+    with open(filename, "r") as original_f:
+        with open(filename + "_", "w") as new_f:
+            new_f.write(new_str)
+            new_f.write(original_f.read())
+    os.rename(filename + "_", filename)
+
+
+def download_decompme(parsed_url: urllib.parse.ParseResult) -> None:
+
+    try:
+        # unique decomp.me identifier for scratches
+        slug = parsed_url.path[1:].split("/")[1]
+
+        response_str = urllib.request.urlopen(f"https://decomp.me/api/scratch/{slug}")
+        response_json = json.load(response_str)
+        func_name = response_json["name"]
+        dirname = create_directory(func_name)
+        urllib.request.urlretrieve(
+            f"https://decomp.me/api/scratch/{slug}/export", f"{dirname}/export.zip"
+        )
+        with zipfile.ZipFile(f"{dirname}/export.zip", "r") as zip_ref:
+            zip_ref.extractall(dirname)
+
+        os.rename(f"{dirname}/code.c", f"{dirname}/base.c")
+        os.rename(f"{dirname}/ctx.c", f"{dirname}/ctx.h")
+        if not os.path.exists("compile.sh"):
+            print(
+                f"Warning!! 'compile.sh' not found in current directory, so it will not be copied to the prepared folder, {dirname}, and this is required by the permuter"
+            )
+        else:
+            shutil.copyfile("compile.sh", f"{dirname}/compile.sh")
+            os.chmod(
+                f"{dirname}/compile.sh",
+                os.stat(f"{dirname}/compile.sh").st_mode | stat.S_IEXEC,
+            )
+
+        prepend_file(f"{dirname}/base.c", '#include "ctx.h"\n')
+        print(
+            f"Success!! Prepared new folder, {dirname}, that is ready for the permuter script! Run with 'python3 permuter.py {dirname} [additional_flags]'"
+        )
+
+    except:
+        print(traceback.format_exc())
+        print("Failed to download from decomp.me")
+        print("Usage:  python3 import.py https://decomp.me/scratch/AbC12")
```

Comment:
sys.argv[0] can be used instead of "python3 import.py", and will then cover using import.py as `./import.py ...` as well. I think this error message is rather bad btw: it's meaningful to emit it if the input URL didn't follow the right syntax -- which would make a lot of sense to check using a regex at the same time as extracting the scratch slug -- but not for other errors.

Reply:
I will make error checking for each case

Reply:
and an appropriate error message for each